airflow/providers/sftp/operators/sftp_batch.py

```diff
@@ -0,0 +1,294 @@
+        force=False,
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.ssh_hook = ssh_hook
+        self.ssh_conn_id = ssh_conn_id
+        self.remote_host = remote_host
+        self.local_files_path = local_files_path
+        self.remote_files_path = remote_files_path
+        self.local_folder = local_folder
+        self.remote_folder = remote_folder
+        self.regexp_mask = regexp_mask
+        self.operation = operation
+        self.confirm = confirm
+        self.create_intermediate_dirs = create_intermediate_dirs
+        self.force = force
+        if not (self.operation.lower() == SFTPOperation.GET or self.operation.lower() == SFTPOperation.PUT):
+            raise TypeError(
+                f"""Unsupported operation value {self.operation},
+                expected {SFTPOperation.GET} or {SFTPOperation.PUT}"""
+            )
+        if not (
+            (
+                self.operation == SFTPOperation.PUT
+                and (
+                    (
+                        isinstance(self.local_folder, str)
+                        and isinstance(self.remote_folder, str)
+                        and local_files_path is None
+                        and remote_files_path is None
+                    )
+                    or (
+                        isinstance(self.local_files_path, list)
+                        and isinstance(self.remote_folder, str)
+                        and self.local_folder is None
+                        and remote_files_path is None
+                        and self.regexp_mask is None
+                    )
+                )
+            )
+            or (
+                self.operation == SFTPOperation.GET
+                and (
+                    (
+                        isinstance(self.local_folder, str)
+                        and isinstance(self.remote_folder, str)
+                        and local_files_path is None
+                        and remote_files_path is None
+                    )
+                    or (
+                        isinstance(self.local_folder, str)
+                        and isinstance(self.remote_files_path, list)
+                        and self.local_files_path is None
+                        and remote_folder is None
+                        and self.regexp_mask is None
+                    )
+                )
+            )
+        ):
+            raise TypeError(
+                """
+                Unsupported argument pool,
+                Possible options for PUT:
+                    1.optional(regexp_mask:str) + local_folder:str + remote_folder:str
+                    2.local_files_path:list + remote_folder:str
+                Possible options for GET:
+                    1.local_folder:str + remote_folder:str + optional(regexp_mask:str)
+                    2.local_folder:str + remote_files_path:list
+                """
+            )
+
+    def execute(self, context: Any) -> str:
```

Comment:
Now that 2.2.3 has been released, the typing of `context` can be:

```python
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from airflow.utils.context import Context

...
def execute(self, context: "Context") -> str:
...
```

Reply:
done